npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -486,8 +486,14 @@ def lockfiles_for_path(lockfiles:, path:)
         def run_checker(path:, version:)
```

Comment:
I'm actually kind of surprised there were no tests to catch this?

Not by peering into the internals, but if this is failing a bunch in prod currently, then maybe could add a unit test that mimics that by running `run_checker` on yarn monorepo and then ensuring it works rather than throwing that error we're seeing?

If this is really hard to write, it may not be worth doing, but if it's straightforward might be worth tacking on?

Reply:
I was struggling how to write a spec for this, I suspect because `npm install` actually succeeds on many yarn repos, it just ignores the yarn lockfile. I guess there are some cases where it does not, but I couldn't easily find an easy case.

Actually while trying to write a spec for this, I discovered the second issue fixed here, for which a test is easier to write.

So I figured this was good with only a test for the latter fix.